post-processing/test_post_processing.py

```diff
@@ -0,0 +1,160 @@
+    if os.path.exists("perflogs"):
+        shutil.rmtree("perflogs", ignore_errors=True)
+    # get test log location path
+    perflog_path = os.path.join(post_processing_dir, "perflogs")
+    # resolve relative path from post-processing directory to access sombrero benchmark
+    sombrero_bench_path = os.path.join(post_processing_dir.parent / "benchmarks/examples/sombrero/sombrero.py")
+
+    # use sombrero example to generate new test perflog file
+    run_benchmark(sombrero_bench_path)
+
+    sombrero_logs = []
+    sombrero_log_re = re.compile(r"^SombreroBenchmark\_\w{8}\.log$")
+    # look for newly generated log files
+    for root, dirs, files in os.walk(perflog_path):
+        sombrero_logs = list(filter(sombrero_log_re.match, files))
+        # if files are found, get file dir path
+        if sombrero_logs:
+            # perflog path with <system>/<partition>
+            perflog_path = root
+
+    sombrero_log_path = ""
+    # check the log files exist
+    if sombrero_logs:
+        # arbitrarily pick one of the log file names
+        sombrero_log_path = os.path.join(perflog_path, sombrero_logs[0])
+
+    # create an incomplete log file
+    sombrero_incomplete_log = "SombreroBenchmarkIncomplete.log"
+    sombrero_incomplete_log_path = os.path.join(perflog_path, sombrero_incomplete_log)
+
+    if os.path.exists(sombrero_log_path):
+        # copy and modify regular log file
+        shutil.copyfile(sombrero_log_path, sombrero_incomplete_log_path)
+
+        REQUIRED_FIELD = "flops_value"
+        required_field_index = 0
+        for line in fileinput.input(sombrero_incomplete_log_path, inplace=True):
+            # find index of column to be removed
+            if fileinput.isfirstline():
+                LOG_FIELDS = line.strip().split("|")
+                required_field_index = LOG_FIELDS.index(REQUIRED_FIELD)
+            # remove a required field
+            new_line = remove_field_from_perflog(line, required_field_index)
+            # inline replacement
+            print(new_line, end="\n")
+
+    yield sombrero_logs, sombrero_log_path, sombrero_incomplete_log_path
+
+    # teardown
+    # clean unnecessary files and folders
+    benchmark_cleanup(remove_test_logs)
+
+# Test that perflog parsing works and information can be extracted to an appropriate DataFrame
+def test_read_perflog(run_sombrero):
+
+    sombrero_logs, sombrero_log_path, sombrero_incomplete_log_path = run_sombrero
+
+    # check the log files exist
+    if sombrero_logs:
+        # check the sombrero log path is valid
+        assert os.path.exists(sombrero_log_path)
+        # check the expected number of log files has been generated
+        assert len(sombrero_logs) ==  4
+    # if the log files haven't been created, something went wrong
+    else:
+        assert False
+
+    # if the incomplete log file hasn't been created, something went wrong
+    if not os.path.exists(sombrero_incomplete_log_path):

```

Comment:
Why not `assert(os.path.exists(sombrero_incomplete_log_path))` instead of the 2-line `if` statement?
Same comment for anywhere else you use `assert False` (but not the check that the exception is thrown below, that one is fine).